addons/website_forum/models/forum_post.py

```diff
@@ -789,6 +790,64 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
     # WEBSITE
     # ----------------------------------------------------------------------
 
+    def _get_microdata(self):
+        """
+        Generate structured data (microdata) for the post.
+
+        Returns:
+            str or None: Microdata in JSON format representing the post, or None
+            if not applicable.
+        """
+        self.ensure_one()
+        # Return if it's not a question.
+        if self.parent_id:
+            return None
+        correct_posts = self.child_ids.filtered(lambda post: post.is_correct)
+        suggested_posts = self.child_ids.filtered(lambda post: not post.is_correct)[:5]
+        # A QAPage schema must have one accepted answer or at least one suggested answer
+        if not suggested_posts and not correct_posts:
+            return None
+
+        structured_data = {
+            "@context": "https://schema.org",
+            "@type": "QAPage",
+            "mainEntity": self._get_structured_data("question")
```

Comment:
When having optional parameters, better to keep it in calls, makes easier to grep / check usage
```suggestion
            "mainEntity": self._get_structured_data(post_type="question")
```